### ISSUE TITLE
[PAID] [$500] [MEDIUM] Feature Request: Auto select the category and/or tag if there is only one option

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#35134]
# Include auto-selection of tags/categories when a single option is available

### Problem
Currently, when a receipt is scanned (or manually created) in a workspace that contains only a single category and/or a single task, the system does not automatically select these options. As a result, users are required to manually choose them even after the receipt is smartscanned. This adds unnecessary steps to the receipt upload > Smartscan process since there's a single option to chose from.

### Solution
Include an auto-selection process to select the category and/or task when a receipt is scanned in a workspace that has only one option available in each category/tag. This would remove the manual selection process and save a few clicks to the user. 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01dd9fec7e03798e66</li>
        <li>Upwork Job ID: 1750912778352738304</li>
        <li>Last Price Increase: 2024-01-26</li>
<li>Automatic offers: </li>
<ul>
<li>jjcoffee | Reviewer | 28135508</li>
<li>dukenv0307 | Contributor | 28135509</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @strepanier03</details>

**Comment by puneetlath** [comment]
@dukenv0307 FYI the issue has been updated. The proposed solution was not correct previously.

**Comment by puneetlath** [comment]
Yes, agreed. I think the confirmation page in any of the request flows -- manual, scan, distance -- should auto-select categories and/or tags if there is only one option for any of them. 

<img width="363" alt="Screenshot 2024-01-26 at 9 17 49 AM" src="https://github.com/Expensify/App/assets/1371996/237c32eb-34a4-4575-a77e-88cfe611e43d">

<img width="367" alt="Screenshot 2024-01-26 at 9 18 04 AM" src="https://github.com/Expensify/App/assets/1371996/1939cd0b-51d2-4332-84ff-6819e77b1288">

<img width="368" alt="Screenshot 2024-01-26 at 9 18 40 AM" src="https://github.com/Expensify/App/assets/1371996/df9f6542-cd4e-4490-939b-d25b07f46d9e">

**Comment by puneetlath** [comment]
@dukenv0307 I'll let @jjcoffee weigh in on that first 😄

**Comment by mavrickdeveloper** [comment]
Hi @puneetlath , i'd like to propose a fix for this 
is this still requires attention or dukenv0307 solution is accepted

**Comment by jjcoffee** [comment]
@dukenv0307's [proposal](https://github.com/Expensify/App/issues/35134#issuecomment-1909769624) sounds good to me!

:ribbon::eyes::ribbon: C+ reviewed

**Comment by jjcoffee** [comment]
@mavrickdeveloper You're always welcome to make a proposal until you see a comment with `C+ reviewed` in it - no need to ask!

**Comment by MariaHCD** [comment]
Removing my assignment since it looks like @puneetlath is the assigned internal engineer here!

**Comment by strepanier03** [comment]
@puneetlath little bump on the proposal review [here](https://github.com/Expensify/App/issues/35134#issuecomment-1914308728).

**Comment by puneetlath** [comment]
Sounds good to me. Thanks for the bump.

**Comment by jjcoffee** [comment]
Not overdue.

**Comment by puneetlath** [comment]
Calm down Melv. @dukenv0307 is working on it.

**Comment by puneetlath** [comment]
Calm down Melv. @dukenv0307 is working on it.

**Comment by jjcoffee** [comment]
@puneetlath Just checking, does it make sense to limit this to only auto-select if a category or tag is required for the workspace?

**Comment by jjcoffee** [comment]
It probably makes sense to hold this for https://github.com/Expensify/App/pull/34983 so that we support multi-level tags here too.

**Comment by puneetlath** [comment]
>@puneetlath Just checking, does it make sense to limit this to only auto-select if a category or tag is required for the workspace?

Good question. I think it makes sense to only auto-select if they are required, like you say.

**Comment by jjcoffee** [comment]
We're just waiting on https://github.com/Expensify/App/pull/34983 here, should be merged today!

**Comment by jjcoffee** [comment]
Nearly there! Just one [small issue](https://github.com/Expensify/App/pull/35862#pullrequestreview-1888569723) to iron out on the PR.

**Comment by jjcoffee** [comment]
@puneetlath Another quick question for you, if you don't mind. Currently categories and tags only display as required on FE if the violations beta is enabled. Should we also respect that here so we only auto-select where the category/tag is required _and_ if the beta is enabled, or do we want to just auto-select regardless of the beta?

**Comment by jjcoffee** [comment]
@puneetlath The BE seems to be inconsistently setting the `required` field for tags (both multiple level and single level). For multiple level tags I consistently see the `required` field when the tag required state is set to false (both on a fresh login, or via pusher update). Setting an optional (non-required) tag to required does keep the `required` field in Onyx if it's via pusher (but won't show on a fresh sign-in). Should we deal with this here or on a separate issue?

**Comment by puneetlath** [comment]
Ok that's good to know. I'll create an issue to look into that. It shouldn't block this PR though, right?

**Comment by puneetlath** [comment]
Testing this on staging and it is properly auto-selecting the Category but not the Tag, even though there is only one. 

<img width="398" alt="Screenshot 2024-03-11 at 2 22 30 PM" src="https://github.com/Expensify/App/assets/1371996/8a01397d-b586-4d50-9bf0-c4fe144eba67">
<img width="382" alt="Screenshot 2024-03-11 at 2 22 39 PM" src="https://github.com/Expensify/App/assets/1371996/c36afc96-39aa-4f6e-ba5d-98fb3e5391e1">

Is that because the policyTags_ object incorrectly says that tags aren't required? Even though the policy_ object says they are? 
<img width="768" alt="Screenshot 2024-03-11 at 2 24 51 PM" src="https://github.com/Expensify/App/assets/1371996/39cf4f91-0427-485c-b503-d22deff66e03">

<img width="282" alt="Screenshot 2024-03-11 at 2 24 30 PM" src="https://github.com/Expensify/App/assets/1371996/13a1ff4e-c8d4-4370-a7da-2fe9b15402ab">

**Comment by jjcoffee** [comment]
@puneetlath Yup that's right, as we [discussed](https://github.com/Expensify/App/pull/35862#issuecomment-1978593064) on the PR it should be working fine for multi-level tags though. Unsure if Yuwen's fix has hit prod yet.

**Comment by trjExpensify** [comment]
@jjcoffee looks like it hit prod 5 days ago: 

<img width="1398" alt="image" src="https://github.com/Expensify/App/assets/16232057/241c8180-3960-4b13-9ead-b60293203c8b">

**Comment by jjcoffee** [comment]
@trjExpensify Hmm can you comment on that issue to say that we're still [seeing](https://github.com/Expensify/App/issues/35134#issuecomment-1989161123) the incorrect `required` prop come through on single-level tags?

**Comment by trjExpensify** [comment]
I've used the Slack thread where we all are: https://expensify.slack.com/archives/C01GTK53T8Q/p1710249218962349?thread_ts=1708732718.173419&cid=C01GTK53T8Q

**Comment by mountiny** [comment]
Noting same as @puneetlath did not work for me for Department tag

**Comment by strepanier03** [comment]
Was OoO yesterday so didn't pay this out. In light of the convo above, should I hold on paying this for now?

**Comment by puneetlath** [comment]
I think we can pay. The outstanding issue is a back-end issue that should get fixed via this PR: https://github.com/Expensify/Auth/pull/10205

**Comment by strepanier03** [comment]
Got it, thanks for weighing in Puneet. I'll move this forward now.

**Comment by strepanier03** [comment]
Payment has been handled for both contributors.

**Comment by strepanier03** [comment]
Closing, but if that was premature feel free to reopen it if needed.
